### PR TITLE
First pass at translation support

### PR DESCRIPTION
**Pull request by JulianGro**
This PR adds limited translation support. Through testing, I deem all changes stable and will continue to work on new functions on https://github.com/JulianGro/vircadia-docs-sphinx/tree/translation-experiment
I am submitting this PR, as the list of changes is growing bigger and I would feel more at ease, getting master up to date with that.

What is changed in this PR:
- update `.gitignore`
    - add `*/build-*` to ignore language builds
    - add `*.mo` to ignore pre-optimized language files
- update `readme.md`
    - remove mention of `Read the Docs`
    - add information on translation builds and contribution
- remove index.rst and have home.rst take its place (since they both had the same content)
    - change `master_doc` to `home.rst`
- add `.rst` to `source_suffix` (this should help with sphinx 3.x.x compatiblity)
- add minimum sphinx version 2.4.4 since we do not support anything else
- add `locale_dirs = ['./locales']` to conf.py to specify where locales are
- fix some warnings
    - fix some malformed tables
    - fix some section title lines
- add german language files
- delete some unused doc pages and images

Please consult the new README.md on how to build specific languages, if you want to test this.

This also influences #11 via commit 7cb700e964faa2d19641ef0b011083c904962381

### CODE REVIEW

File: docs/source/locales/de/LC_MESSAGES/contribute.po [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#
+#, fuzzy
+msgid ""
+msgstr ""
+"Project-Id-Version: Vircadia Docs \n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2020-06-25 03:02+0200\n"
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
```
Comment (ctrlaltdavid) [inline docs/source/locales/de/LC_MESSAGES/contribute.po:15]: Is the language team URL correct?

File: docs/source/locales/de/LC_MESSAGES/contribute.po [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# SOME DESCRIPTIVE TITLE.
```
Comment (ctrlaltdavid) [inline docs/source/locales/de/LC_MESSAGES/contribute.po:1]: Where does "SOME DESCRIPTIVE TITLE" come from? Is it OK to leave this placeholder text?

File: docs/source/locales/de/LC_MESSAGES/contribute.po [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+# Copyright (C) 2019, High Fidelity. © Copyright 2020, Vircadia.
+# This file is distributed under the same license as the Vircadia Docs
+# package.
+# FIRST AUTHOR <EMAIL@ADDRESS>, 2020.
+#
+#, fuzzy
```
Comment (ctrlaltdavid) [inline docs/source/locales/de/LC_MESSAGES/contribute.po:7]: "fuzzy"?

File: README.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Overview of Vircadia's Documentation Tools
 
-For Vircadia's main documentation system, we use **Sphinx** to generate it, and **Read the Docs** to publish/host it. GitHub is a helpful middleman and stores all of the docs.
+For Vircadia's main documentation system, we use **Sphinx** to generate it, and a webserver to publish/host it. GitHub is a helpful middleman and stores all of the docs.
```
Comment (ctrlaltdavid) [inline README.md:3]:
```suggestion
For Vircadia's main documentation system, we use **Sphinx** to generate it, and a web server to publish/host it. GitHub is a helpful middleman and stores all of the docs.
```

File: docs/source/locales/de/LC_MESSAGES/contribute.po (hunk 1 above)
Comment (JulianGro) [reply docs/source/locales/de/LC_MESSAGES/contribute.po:15]: No that is a placeholder URL that comes from sphinx. I am not sure about changing the stuff like this, as it is automatically generated and not really used anywhere. Just feels like wasted time looking into that stuff.

File: docs/source/locales/de/LC_MESSAGES/contribute.po (hunk 3 above)
Comment (JulianGro) [reply docs/source/locales/de/LC_MESSAGES/contribute.po:7]: This is used by the translation softwares to mark strings that need a look at. Either the source string has been changed, or there is not a proper translation yet.

File: docs/source/locales/de/LC_MESSAGES/contribute.po (hunk 2 above)
Comment (JulianGro) [reply docs/source/locales/de/LC_MESSAGES/contribute.po:1]: This was generated by sphinx. Trying to find out how to change this seems like a waste of time.

**Pull request merged**